public/app/plugins/panel/piechart/PieChartPanel.tsx

```diff
@@ -127,6 +135,15 @@ function getLegend(props: Props, displayValues: FieldDisplay[]) {
   );
 }
 
+function hasFrames(fieldDisplayValues: FieldDisplay[]) {
+  for (let fd of fieldDisplayValues) {
+    if (fd.view && fd.view.dataFrame.length > 0) {
+      return true;
+    }
+  }
+  return false;
+}
+
```

Comment:
Do we need this function? Can't we simply rely on either data frames existence or fields compatibility with the vis? Something similar to what we do in the Bar chart: https://github.com/grafana/grafana/blob/8b80d2256dabbb29bb2beb3e97773789af61b22f/public/app/plugins/panel/barchart/BarChartPanel.tsx#L16

Reply:
Time series relies on data.series being > 0. In this case we actually get series ( DataFrame[] ) but the array is empty. It's fine to display this in the time series as it will just be a graph with no points in it, and that is probably the expected behavior. But for the PieChart that's more like the Gauge plugin it looks odd when we don't get any values, that's why I wanna check that there's some data to display.